pkg/resourcemanager/cosmosdb/sqldatabase/cosmosdbsql.go

```diff
@@ -0,0 +1,275 @@
+			errhelp.InvalidFailoverGroupRegion,
+		}
+		return errhelp.IsErrorFatal(err, append(allowedErrors, notFoundErrorCodes...), unrecoverableErrors)
+	}
+	instance.Status.SetProvisioning("Resource request successfully submitted to Azure")
+	instance.Status.PollingURL = future.PollingURL()
+
+	// Need to poll the polling URL, so not done yet!
+	return false, nil
+}
+
+func (m *AzureCosmosDBSQLDatabaseManager) Delete(ctx context.Context, obj runtime.Object, opts ...resourcemanager.ConfigOption) (bool, error) {
+	options := &resourcemanager.Options{}
+	for _, opt := range opts {
+		opt(options)
+	}
+
+	instance, err := m.convert(obj)
+	if err != nil {
+		return false, err
+	}
+
+	cosmosDBSQLDatabaseClient, err := cosmosdb.GetCosmosDBSQLDatabaseClient(m.Creds)
+	if err != nil {
+		return false, errors.Wrapf(err, "failed to create cosmos DB SQL database client")
+	}
+
+	_, err = cosmosDBSQLDatabaseClient.DeleteSQLDatabase(ctx, instance.Spec.ResourceGroup, instance.Spec.CosmosDBAccount, instance.ObjectMeta.Name)
+	if err != nil {
+		instance.Status.Message = err.Error()
+		azerr := errhelp.NewAzureError(err)
+
+		// these errors are expected
+		ignore := []string{
+			errhelp.AsyncOpIncompleteError,
+		}
+
+		// this means the thing doesn't exist
+		finished := []string{
+			errhelp.ResourceNotFound,
+			errhelp.ResourceGroupNotFoundErrorCode,
+		}
+
+		if helpers.ContainsString(ignore, azerr.Type) {
+			return true, nil
+		}
+
+		if helpers.ContainsString(finished, azerr.Type) {
+			return false, nil
+		}
+		instance.Status.Message = fmt.Sprintf("delete failed with: %s", err.Error())
+		return false, err
+	}
+
+	// Not done yet, keep trying
+	return false, nil
+}
+
+func (m *AzureCosmosDBSQLDatabaseManager) GetParents(obj runtime.Object) ([]resourcemanager.KubeParent, error) {
+	instance, err := m.convert(obj)
+	if err != nil {
+		return nil, err
+	}
+
+	return []resourcemanager.KubeParent{
+		{
+			Key: types.NamespacedName{
+				Namespace: instance.Namespace,
+				Name:      instance.Spec.CosmosDBAccount,
+			},
+			Target: &v1alpha1.CosmosDB{},
+		},
+		{
+			Key: types.NamespacedName{
+				Namespace: instance.Namespace,
+				Name:      instance.Spec.ResourceGroup,
+			},
+			Target: &v1alpha1.ResourceGroup{},
+		},
+	}, nil
+}
+
+func (m *AzureCosmosDBSQLDatabaseManager) GetStatus(obj runtime.Object) (*v1alpha1.ASOStatus, error) {
+	instance, err := m.convert(obj)
+	if err != nil {
+		return nil, err
+	}
+
+	return &instance.Status, nil
+}
+
+/*
+ * Helper functions
+ */
+
+func (m *AzureCosmosDBSQLDatabaseManager) convert(obj runtime.Object) (*v1alpha1.CosmosDBSQLDatabase, error) {
+	db, ok := obj.(*v1alpha1.CosmosDBSQLDatabase)
+	if !ok {
+		return nil, fmt.Errorf("failed type assertion on kind: %s", obj.GetObjectKind().GroupVersionKind().String())
+	}
+	return db, nil
+}
+
+func makeTags(tags map[string]string) map[string]*string {
+	result := make(map[string]*string)
+	for key, value := range tags {
+		result[key] = &value
```

Comment:
Is this going to end up with all the values being the same?
```suggestion
		localValue := value
		result[key] = &localValue
```

Reply:
Ahh! I think you're right.

Isn't there a linter for this? I thought @Porges had enabled it.

Reply:
Yeah, I think there was! Maybe it's only running on the codegen stuff?

Reply:
Yeah it's only running on the codegen stuff right now it looks like (because it's run through the taskfile which is only invoked on the hack/* stuff).